main.nf

```diff
@@ -1,130 +1,575 @@
+if (params.config_profile_description) summary['Config Description'] = params.config_profile_description
+if (params.config_profile_contact)     summary['Config Contact']     = params.config_profile_contact
+if (params.config_profile_url)         summary['Config URL']         = params.config_profile_url
+if (params.email || params.email_on_fail) {
+  summary['E-mail Address']    = params.email
+  summary['E-mail on failure'] = params.email_on_fail
+  summary['MultiQC maxsize']   = params.maxMultiqcEmailFileSize
+}
+log.info summary.collect { k,v -> "${k.padRight(18)}: $v" }.join("\n")
+log.info "-\033[2m--------------------------------------------------\033[0m-"
+
+// Check the hostnames against configured profiles
+checkHostname()
+
+def create_workflow_summary(summary) {
+    def yaml_file = workDir.resolve('workflow_summary_mqc.yaml')
+    yaml_file.text  = """
+    id: 'nf-core-neutronstar-summary'
+    description: " - this information is collected when the pipeline is started."
+    section_name: 'nf-core/neutronstar Workflow Summary'
+    section_href: 'https://github.com/nf-core/neutronstar'
+    plot_type: 'html'
+    data: |
+        <dl class=\"dl-horizontal\">
+${summary.collect { k,v -> "            <dt>$k</dt><dd><samp>${v ?: '<span style=\"color:#999999;\">N/A</a>'}</samp></dd>" }.join("\n")}
+        </dl>
+    """.stripIndent()
+
+   return yaml_file
+}
+
 
 Channel
-    .value([p.id, p.fastqs])
-    .into { supernova_input; longranger_input }
+    .from(samples)
+    .set { supernova_input }
+
+if (params.full_output) {
+    process supernova_full {
+        tag "${id}"
+        label "supernova"
+        publishDir "${params.outdir}/supernova/", mode: 'copy'
+
+        input:
+        set val(id), val(fastqs), val(tenx_options), val(supernova_options) from supernova_input
+
+
+        output:
+        set val(id), file("${id}/*") into supernova_results, supernova_results2
+
+        script:
+        """
+        supernova run --id=${id} --fastqs=${fastqs} ${tenx_options} ${supernova_options}
+        """
+    }
 
-process supernova {
+} else {
+    process supernova {
+        tag "${id}"
+        label "supernova"
+        publishDir "${params.outdir}/supernova/", mode: 'copy'
+
+        input:
+        set val(id), val(fastqs), val(tenx_options), val(supernova_options) from supernova_input
+
+        output:
+        set val(id), file("${id}_supernova") into supernova_results, supernova_results2
+
+        script:
+        """
+        supernova run --id=${id} --fastqs=${fastqs} ${tenx_options} ${supernova_options}
+        rsync -rav --include="_*" --include="*.tgz" --include="outs/" --include="outs/*.*"  --include="assembly/" --include="stats/***" --include="logs/***" --include="a.base/" --include="a.base/" --include="a.hbx" --include="a.inv" --include="final/***" --include="gang" --include="micro"  --include="a.hbx" --include="a.inv" --include="final/***" --exclude="*" "${id}/" ${id}_supernova
```

Comment:
It could be nice to break this command over at least a few lines so that it's a bit easier to read... (maybe one per `--include` so that they're vertically stacked?)

Reply:
Also would be good to add a comment line above the script explaining why you need the `rsync` command..